daffodil-core/src/main/scala/org/apache/daffodil/dsom/Facets.scala

```diff
@@ -160,7 +162,7 @@ trait Facets { self: Restriction =>
           // The XSD numeric character entity &#xE000; can be used to match ASCII NUL
           // (char code 0).
           //
-          val remapped = XMLUtils.remapPUAToXMLIllegalCharacters(v)
+          val remapped: String = XMLUtils.remapPUAToXMLIllegalCharacters(v)
```

Comment:
I'm curious about these explicit types (lines 133, 136, 165).  What was the reason you added these explicit types?  I thought the Scala compiler inferred types for local values fairly well although sometimes IntelliJ IDEA gets a little confused itself (is that why?).

Reply:
Yes, I needed these in order to figure out what was going on here. Type aliases were confusing IDEA. I tend to leave in that sort of thing when it's needed to improve understanding, why take it back out.